RUFAS/biophysical/manure/separators/separator.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from RUFAS.biophysical.manure.processor import Processor
+from RUFAS.current_day_conditions import CurrentDayConditions
+from RUFAS.data_structures.animal_to_manure_connection import ManureStream
+from RUFAS.time import Time
+
+
+@dataclass
+class SeparatorConfig:
+    """
+    Base configuration for a manure separator.
+
+    Attributes
+    ----------
+    water_efficiency : float
+        The efficiency of the separator in removing water from the manure.
+    ammoniacal_nitrogen_efficiency : float
+        The efficiency of the separator in removing ammoniacal nitrogen from the manure.
+    nitrogen_efficiency : float
+        The efficiency of the separator in removing nitrogen from the manure.
+    phosphorus_efficiency : float
+        The efficiency of the separator in removing phosphorus from the manure.
+    potassium_efficiency : float
+        The efficiency of the separator in removing potassium from the manure.
+    ash_efficiency : float
+        The efficiency of the separator in removing ash from the manure.
+    non_degradable_volatile_solids_efficiency : float
+        The efficiency of the separator in removing non-degradable volatile solids from the manure.
+    degradable_volatile_solids_efficiency : float
+        The efficiency of the separator in removing degradable volatile solids from the manure.
+    total_solids_efficiency : float
+        The efficiency of the separator in removing total solids from the manure.
+    """
+
+    water_efficiency: float
+    ammoniacal_nitrogen_efficiency: float
+    nitrogen_efficiency: float
+    phosphorus_efficiency: float
+    potassium_efficiency: float
+    ash_efficiency: float
+    non_degradable_volatile_solids_efficiency: float
+    degradable_volatile_solids_efficiency: float
+    total_solids_efficiency: float
+
+
+class Separator(Processor):
+    """
+    A manure processor that separates solids from liquids.
+
+    Parameters
+    ----------
+    config : SeparatorConfig
+        Configuration for the separator.
+    is_housing_emissions_calculator : bool
+        Indicates if a Processor calculates housing emissions.
+
+    Attributes
+    ----------
+    config : SeparatorConfig
+        Configuration for the separator.
+    is_housing_emissions_calculator : bool
+        If true, processor will only accept ManureStreams with non-None PenManureData, if false then vice versa.
+
+    """
+
+    def __init__(self, config: SeparatorConfig, is_housing_emissions_calculator: bool = False) -> None:
+        """Initializes a new Separator."""
+        super().__init__(is_housing_emissions_calculator)
+        self.config: SeparatorConfig = config
+        self.held_manure: ManureStream | None = None
```

Comment:
```suggestion
    def __init__(self, config: SeparatorConfig) -> None:
        """Initializes a new Separator."""
        super().__init__(is_housing_emissions_calculator=False)
        self.config: SeparatorConfig = config
        self.held_manure: ManureStream | None = None
```
I think we can be reasonably confident that there will never be a separator that calculates housing emissions.